File: c++_solutions/3100.cpp
class Solution {
public:
    int maxBottlesDrunk(int numBottles, int numExchange) {
        int drunk = 0 ;
        while (numBottles > 0){
            if (numBottles >= numExchange){
                drunk += numExchange ; 
                numBottles -= numExchange ; 
                numBottles += 1 ;
                numExchange += 1 ;
            }
            else{
                drunk += numBottles ; 
                numBottles = 0;
            }
        }
        return drunk ;
    }
};

File: rust_solutions/3100.rs
impl Solution {
    pub fn max_bottles_drunk(num_bottles: i32, num_exchange: i32) -> i32 {
        let mut nb = num_bottles ; 
        let mut ne = num_exchange ;
        let mut drunk : i32 = 0 ;
        while nb  > 0 {
            if nb >= ne {
                drunk += ne ; 
                nb -= ne ;
                nb += 1 ;
                ne += 1 ;   
            }
            else{
                drunk += nb ; 
                nb = 0 ;
            }
        }
        drunk
    }
}
